src/WSRequestHandler_MediaControl.cpp

```diff
@@ -0,0 +1,396 @@
+	}
+
+	OBSSourceAutoRelease source = obs_get_source_by_name(sourceName.toUtf8());
+	if (!source) {
+		return request.failed("specified source doesn't exist");
+	}
+
+	obs_source_media_set_time(source, timestamp);
+	return request.success();
+}
+
+/**
+* Scrub media using a supplied offset. Supports ffmpeg and vlc media sources (as of OBS v25.0.8)
+* Note: Due to processing/network delays, this request is not perfect. The processing rate of this request has also not been tested.
+*
+* @param {String} `sourceName` Source name.
+* @param {int} `timeOffset` Millisecond offset (positive or negative) to offset the current media position.
+*
+* @api requests
+* @name ScrubMedia
+* @category media control
+* @since 4.9.0
+*/
+RpcResponse WSRequestHandler::ScrubMedia(const RpcRequest& request) {
+	if (!request.hasField("sourceName") || !request.hasField("timeOffset")) {
+		return request.failed("missing request parameters");
+	}
+
+	QString sourceName = obs_data_get_string(request.parameters(), "sourceName");
+	int64_t timeOffset = (int64_t)obs_data_get_int(request.parameters(), "timeOffset");
+	if (sourceName.isEmpty()) {
+		return request.failed("invalid request parameters");
+	}
+
+	OBSSourceAutoRelease source = obs_get_source_by_name(sourceName.toUtf8());
+	if (!source) {
+		return request.failed("specified source doesn't exist");
+	}
+
+	int64_t newTime = obs_source_media_get_time(source) + timeOffset;
+	if (newTime < 0) {
+		newTime = 0;
+	}
+
+	obs_source_media_set_time(source, newTime);
+	return request.success();
+}
+
+/**
+* Get the current playing state of a media source. Supports ffmpeg and vlc media sources (as of OBS v25.0.8)
+*
+* @param {String} `sourceName` Source name.
+*
+* @return {String} `mediaState` The media state of the provided source. States: `none`, `playing`, `opening`, `buffering`, `paused`, `stopped`, `ended`, `error`, `unknown`
+*
+* @api requests
+* @name GetMediaState
+* @category media control
+* @since 4.9.0
+*/
+RpcResponse WSRequestHandler::GetMediaState(const RpcRequest& request) {
+	if (!request.hasField("sourceName")) {
+		return request.failed("missing request parameters");
+	}
+
+	QString sourceName = obs_data_get_string(request.parameters(), "sourceName");
+	if (sourceName.isEmpty()) {
+		return request.failed("invalid request parameters");
+	}
+
+	OBSSourceAutoRelease source = obs_get_source_by_name(sourceName.toUtf8());
+	if (!source) {
+		return request.failed("specified source doesn't exist");
+	}
+
+	OBSDataAutoRelease response = obs_data_create();
+	obs_data_set_string(response, "mediaState", getSourceMediaState(source).toUtf8());
+
+	return request.success(response);
+}
+
+/**
+* List the media state of all media sources (vlc and media source)
+*
+* @return {Array<Object>} `mediaSources` Array of sources
+* @return {String} `mediaSources.*.sourceName` Unique source name
+* @return {String} `mediaSources.*.sourceTypeId` Non-unique source internal type (a.k.a `ffmpeg_source` or `vlc_source`)
```

Comment:
Type id is called "source kind" everywhere else so I'd rather stay consistent with that.
```suggestion
* @return {String} `mediaSources.*.sourceKind` Unique source internal type (a.k.a `ffmpeg_source` or `vlc_source`)
```